src/Microsoft.AspNetCore.Mvc.Formatters.Json/MvcJsonOptionsExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    public static class MvcJsonOptionsExtensions
+    {
+        /// <summary>
+        /// Applies camelCasing to property names and dictionary keys.
+        /// </summary>
+        public static MvcJsonOptions UseCamelCasing(this MvcJsonOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+            
+            options.SerializerSettings.ContractResolver = new CamelCasePropertyNamesContractResolver();
```

Comment:
What's the reason for using `CamelCasePropertyNamesContractResolver` instead of `DefaultContractResolver`?

Reply:
I followed the reasoning in this [comment](https://github.com/aspnet/Mvc/issues/7256#issuecomment-360690993).
> That is, as the developer as soon as I configure camelCasing explicitly, I know that EVERYTHING going to client would be camelCased, and I take the responsibility.

Reply:
> I know that EVERYTHING going to client would be camelCased

But if you've explicitly named some properties, would you expect them to be overridden when calling a method called `UseCamelCase`?

I guess what we're coming back to is that "camel case" could mean a lot of different things. Forcing the developer to be explicit about the behaviors they want, or at least documenting it properly quickly leads to confusion and subtle bugs.

Reply:
BTW, my comment was about using `DefaultContractResolver` with `CamelCaseNamingStrategy` instead of `CamelCasePropertyNamesContractResolver`. That would also allow you to reuse the shared contract resolver like my comment below.

Reply:
I personally would expect everything to change to camelCase if I state it in configuration despite the fact that I've annotated some properties differently. However, the fact that we're having this discussion clearly shows that, as it stands now, the current implementation will lead to confusion. I'll go with your approach.